### ISSUE TITLE
[Bug]: possible regression in axis ticks handling in matplotlib 3.6.0rc2

### ISSUE THREAD

**Issue by neutrinoceros** [matplotlib/matplotlib#23806]
### Bug summary

While running image tests for yt, I noticed a change of behaviour that bisects to 
[2357c92d87d96d519c8470776e76180e71663d0b](https://github.com/matplotlib/matplotlib/commit/2357c92d87d96d519c8470776e76180e71663d0b) (PR #22587)
and that I'm not 100% sure is intentional, where parameters set with `Axis.tick_params` are forgotten after a call to `Axes.cla`.

In yt we use persistent `Axis` instances inside container classes; we typically set some `tick_params` during instantiation of the container, and defer rendering as much as possible, at which point `Axis` instances may be cleared before the plot is drawn again.

I think this can easily be fixed in yt, but I wanted to check with you wether this change was intented or not.

### Code for reproduction

```python
import matplotlib as mpl
import matplotlib.pyplot as plt

fig, ax = plt.subplots()
ax.tick_params(which="both", axis="both", direction="in", top=True, right=True)
ax.cla()
fig.savefig(f"/tmp/{mpl.__version__}.png")
```

### Actual outcome

![3 6 0 dev3625+g25b39d41d3](https://user-images.githubusercontent.com/14075922/188493019-33cfd10b-c1b8-46d7-b6c2-7a9be5b566ef.png)

### Expected outcome

with matplotlib 3.5
![3 5 0rc2 dev89+g6e95e7bf54](https://user-images.githubusercontent.com/14075922/188493230-b991a409-cf0c-401a-8c4a-c5b83d19cd2a.png)

### Operating system

N/A

### Matplotlib Version

3.6 VS 3.5

### Matplotlib Backend

MacOSX (but also seeing this on a Linux CI server)

### Python version

3.10.4

### Jupyter version

N/A

### Installation

pip

**Comment by timhoffm** [comment]
Thanks for bringing this up. I think we need a discussion what we mean by `clear()`.

I don't have the full history present and think we might have been inconsistent here.

From a conceptual point of view. IMHO `clear()` should reset all stylistic attributes (including `tickdir`, which feels the same category as linewdith to me). https://github.com/matplotlib/matplotlib/pull/22587 is a rewrite to explicitly not reset tick and ticklabel visibility. This is because visibility is more structural than stylistic. It's e.g. set by shared layouts and resetting visibility would make `clear()` unusable for shared Axes. Whether that's the clean way would be a separate discussion, but since we don't have a persistent concept of layout connected to tick visiblity, not resetting visibility is the only viable approach. I don't think these arguments hold for other tick properties.

To be checked: We might have been inconsistent and even changed behavior unintendedly or at least without warning. So independent on what is a good behavior, we need to check the past and current behavior and also make sure we have consistency, or if that is not possible at least a documented behavior over time.

**Comment by jklymak** [comment]
My intuition is that clearing the axes removes artists, puts the axes back to its default axes limits, and clears label. Ie all data added to the axes would be removed. 

OTOH, axes styling - tickdir, tick length, which spines are shown, spine styling, etc would not be reset as these are independent of the data on the axes.  That is what I believe the behaviour of Matlab is, and I think it is the behaviour Matplotlib has had until recently.

**Comment by neutrinoceros** [comment]
Note that I opened #23808 to restore the behaviour of matplotlib 3.5, if you guys want to stay conservative.

**Comment by tacaswell** [comment]
I'm inclined to be conservative and go back to the 3.5 behavior rather than adding a behavior change note at this point.

I think it is better be consistent through time but inconsistent through the API (unless we can make the API fully consistent in one go or at least have a clearly articulated plan of where we are going).

**Comment by timhoffm** [comment]
> I'm inclined to be conservative and go back to the 3.5 behavior rather than adding a behavior change note at this point.

Agreed, but still we need to consider two which properties this should apply. For example, 3.5 also keeps `tick_params(... width=10)` under `clear()` whereas 3.6rc does not. I suppose this holds for a lot of the tick_params.

**Comment by QuLogic** [comment]
This was fixed by #23834; discussion for what to do long term is in #23839.
